### PR TITLE
3DS 001-* Support Code Docs (and matching WiiU 101-* Codes)

### PR DESCRIPTION
**Pull request by bladetyr**
**Note:** Issues are disabled for this repository. I was going to make an issue first and then link it to this PR, but I was unable to. For that reason, I'm PR-ing a small amount of work to make sure this is in the desired formatting.

Let me know if any corrections need to be made! As far as I am aware, the attributes go like:
- NAME: Only applies to NEX or Spider modules.
- DESCRIPTION: Brief description of the issue.
- MESSAGE: Actual error message that gets shown with the error code on the console.
- LINK: Link to official Nintendo support docs.

### ADDED:
- Links to any _official_ Nintendo documentation on the error.
- Any known error messages (some were pulled from user photos in the support channels from the Pretendo Discord. I only used the ones already in English).
- Any known fixes found in the support channels/online.
- Descriptions for most errors. Some weren't detailed enough for me to make one.

### DEMO:
image

### CODE REVIEW

**Comment by jonbarrow** [comment]
> Issues are disabled for this repository

I was not aware of this, they have been enabled now. Sorry about that. Let me know if any other repos have these disabled when they otherwise should

> NAME: Only applies to NEX or Spider modules.

The error name applies to all modules, if error names are known. You can see here that many modules have error names https://github.com/PretendoNetwork/Bandwidth/tree/b29d668396a9ac689d7591b77ce1a14903ed5301/src/console-errors/wiiu/support-codes/codes

> DESCRIPTION: Brief description of the issue.

This is the module description, not the error description

> MESSAGE: Actual error message that gets shown with the error code on the console.

Correct :+1: 

> LINK: Link to official Nintendo support docs.

Sort of. It's a link to a help page. It doesn't HAVE to be an official Nintendo document though. For example we have support pages for some errors on our website (https://pretendo.network/docs/errors/SYS-102-2482). Most of the time the links will be official links however, unless it's an error code we've created, because our own support pages are pretty incomplete. Eventually I'd like to move all links to our website however

> DEMO:

About this demo specifically, it should be noted that it seems like most, if not all, 3DS errors line up with Wii U errors in the same modules. The 3DS errors simply use `0XX` while the Wii U uses `1XX`

So for example the `001-0112` error in your demo would be the same as `101-0112` on the Wii U (https://github.com/PretendoNetwork/Bandwidth/blob/b29d668396a9ac689d7591b77ce1a14903ed5301/src/console-errors/wiiu/support-codes/codes/101-fp.js#L25)

It should be noted however that we will likely be trashing the current implementation of the support codes in favor of something JSON based, as to support translations of errors and fixes into multiple languages. A draft PR is open here https://github.com/PretendoNetwork/Bandwidth/pull/15

**Comment by bladetyr** [comment]
Understood; thanks for the clarification on the attributes too! I'll update the WiiU codes just in case you'd like this PR as a "just for now" type of thing until you can refactor to something else.

**Comment by bladetyr** [comment]
All done. Couldn't find anything online to verify that 101-0803 was the exact same as the 3DS 000-0803 but included it anyway based on what you said earlier. The WiiU error name seems to suggest they are the same.

Obligatory demo image:
image

File: src/console-errors/3ds/support-codes/codes/001-frd.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-	'0725': new SupportCode('Missing name', 'Missing description', 'Friend Registration is restricted by\nParental Controls', 'Missing fix', 'Missing link'),
-	'0726': new SupportCode('Missing name', 'Missing description', 'You have not yet configured your\nnotification settings for friends.\n\nConfigure these settings in the\nfriend list', 'Missing fix', 'Missing link'),
-	'0803': new SupportCode('Missing name', 'Missing description', 'Missing message', 'Missing fix', 'Missing link'),
-	'0811': new SupportCode('Missing name', 'Missing description', 'The server is currently undergoing\nmaintenance. We apologize for any\ninconvenience. Please try again later.\n\nTo learn more about maintenance,\nvisit support.nintendo.com', 'Missing fix', 'Missing link'),
-};
+const SupportCode = require('../../../common/support-code');
+
+module.exports = {
+	'0102': new SupportCode('Missing name', 'Likely a Pretendo server issue', 'Missing message.', 'Try double checking your internet connection or wait.', 'https://en-americas-support.nintendo.com/app/answers/detail/a_id/17043/~/error-code%3A-001-0102#:~:text=This%20error%20code%20indicates%20you,a%20poor%20wireless%20network%20environment.'),
```
Comment (jonbarrow) [inline src/console-errors/3ds/support-codes/codes/001-frd.js:4]: Adding the period to the end of `Missing message` breaks consistency with the other strings. The period itself is fine, but for consistency *all* other strings, across all modules, should be updated in the same way. I'm not sure if this is something you want to cover in this PR, or just revert the change

File: src/console-errors/wiiu/support-codes/codes/101-fp.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ module.exports = {
 	'0109': new SupportCode('CORE_INVALID_INDEX', 'Missing description', 'Missing message', 'Missing fix', 'Missing link'),
 	'0110': new SupportCode('CORE_OUT_OF_MEMORY', 'Missing description', 'Missing message', 'Missing fix', 'Missing link'),
 	'0111': new SupportCode('CORE_INVALID_ARGUMENT', 'Missing description', 'Missing message', 'Missing fix', 'Missing link'),
-	'0112': new SupportCode('CORE_TIMEOUT', 'Missing description', 'Unable to connect to the server.\n\nPlease try again later.\n\nIf the problem persists, please\nmake a note of the error code and\nvisit support.nintendo.com.', 'Missing fix', 'Missing link'),
+	'0112': new SupportCode('CORE_TIMEOUT', 'Connection timeout', 'Unable to connect to the server.\nPlease try again later', 'Verify your wifi is working. If it is\nworking, then this is a serverside error.\nPlease wait and try again later.', 'Missing link'),
```
Comment (jonbarrow) [inline src/console-errors/wiiu/support-codes/codes/101-fp.js:25]: Why was the message changed here? The message should be as it appears on the console. These were extracted from the consoles MSBT files, so they should already be how the error displays on the console?

Comment (bladetyr) [reply src/console-errors/wiiu/support-codes/codes/101-fp.js]: WiiU description was changed in error, sorry about that. It's fixed now!
